projects/discussion-ui/src/lib/components/tag-all-discussion/tag-all-discussion.component.ts

```diff
@@ -0,0 +1,138 @@
+import { Component, OnInit } from '@angular/core'
+import { NSDiscussData } from '../../models/discuss.model'
+import { Router, ActivatedRoute } from '@angular/router'
+import { DiscussionService } from '../../services/discussion.service';
+/* tslint:disable */
+import _ from 'lodash'
+import { Subscriber, Subscription } from 'rxjs';
+import { ConfigService } from '../../services/config.service';
+import * as CONSTANTS from './../../common/constants.json';
+import { DiscussUtilsService } from '../../services/discuss-utils.service';
+
+@Component({
+  selector: 'lib-tag-all-discussion',
+  templateUrl: './tag-all-discussion.component.html',
+  styleUrls: ['./tag-all-discussion.component.scss']
+})
+export class TagAllDiscussionComponent implements OnInit {
+
+  tagName!: any
+  similarPosts!: any
+  queryParam: any
+  fetchSingleCategoryLoader = false
+  currentActivePage!: any
+  defaultError = 'Something went wrong, Please try again after sometime!'
+  pager = {}
+  paginationData!: any
+  fetchNewData = false
+  paramsSubscription: Subscription;
+  getParams: any;
+  cIds: any;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private discussService: DiscussionService, public activatedRoute: ActivatedRoute,
+    private configService: ConfigService,
+    private discussUtils: DiscussUtilsService
+  ) { }
+
+  ngOnInit() {
+
+    this.getParams = this.configService.getConfig()
+    this.cIds = _.get(this.getParams, 'categories')
+
+    this.activatedRoute.queryParams.subscribe((params) => {
+      this.tagName = params.tagname
+    })
+    if (this.cIds.result.length) {
+      this.fetchContextBasedTagDetails(this.tagName, this.cIds, this.currentActivePage)
+    } else {
+      this.currentActivePage = 1
+      this.fetchSingleTagDetails(this.tagName, this.currentActivePage)
```

Comment:
When the variable is private level, then why are we passing as functional variables to other func?
It can directly access the private level variables, right?